### PR TITLE
HTML Code Style Guide

### PR DESCRIPTION
**Pull request by mike-sheppard**
adding in the HTML code style guide

### CODE REVIEW

**Comment by ericescalante** [comment]
Looks great, I sometimes forget about boolean attributes :+1:

File: coding-styles/html/README.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+1. [Reducing markup](#reducing-markup)
+1. [JavaScript generated markup](#javascript-generated-markup)
+1. [Editor preferences](#editor-preferences)
+
+
+## Syntax
++ Use two spaces, not tabs
++ Nested elements should be indented once (two spaces).
++ Always use double quotes, never single quotes, on attributes.
++ Don't include a trailing slash in self-closing elements—the [HTML5 spec](http://dev.w3.org/html5/spec-author-view/syntax.html#syntax-start-tag) says they're optional.
```
Comment (minsikzzang) [inline coding-styles/html/README.md:23]: Can you please add a few good / bad examples for some of syntax rules? like

``` HTML
// good
<div class="test">

// bad
<div class='test' />
```

File: coding-styles/html/README.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
++ `class`
++ `id`, `name`
++ `data-*`
++ `src`, `for`, `type`, `href`, `value`
++ `title`, `alt`
++ `role`, `aria-*`
+
+Classes make for great reusable components, so they come first. Ids are more specific and should be used sparingly (e.g., for in-page bookmarks), so they come second.
+
+```html
```
Comment (minsikzzang) [inline coding-styles/html/README.md:143]: Good / bad examples please here also :)

**Comment by minsikzzang** [comment]
LGTM, but would be great if we can have a few more examples for the rules.

**Comment by minsikzzang** [comment]
:+1:

**Pull request merged**